### PR TITLE
Proactively fix <0 Temp Bug

### PR DESCRIPTION
**Pull request by Litemage**
# Overview

When looking into the MAX31856 library code for #6 , I noticed that we were not properly accounting for the conversion between *sign-magnitude* and *twos-compliment* signed integer formats when reading from MAX31856 registers. 

Because the math works out, any internal temperatures greater than zero would not have exhibited any strange behaviors, but using this board in an environment where the temperature would be less than zero would result in erroneous values. 

# What's In This PR

This PR implements a fix for the above by adding two functions for converting from signed-magnitude and twos compliment, as well as implementing them in the sample function.

### CODE REVIEW

File: src/PWFusion_MAX31856.cpp [hunk 1]
```diff
@@ -179,22 +187,28 @@ void MAX31856::config(Tc_Type TC_TYPE, uint8_t FILT_FREQ, uint8_t AVG_MODE, Max3
...
-   rawCJTemp  = (int16_t)(uint16_t)readByte(REG_CJTH) << 8; // MSB, left shift 8
-   rawCJTemp |= (int16_t)(uint16_t)readByte(REG_CJTL);              // LSB read
-   // now save sign, shift right 2 to align to type (see datasheet, pg 24 for reg packing)
-	rawCJTemp >>= 2;
+   rawCJReg  = (int16_t)(uint16_t)readByte(REG_CJTH) << 8; // MSB, left shift 8
+   rawCJReg |= (int16_t)(uint16_t)readByte(REG_CJTL);              // LSB read
+
+   // convert to 16-bit sign-magnitude value
+   rawCJReg = (rawCJReg >> 2) & 0b1001111111111111;
+   rawCJTemp = smtotc_16(rawCJReg);
```
Comment (CodingWithFusion) [inline src/PWFusion_MAX31856.cpp]: What do you think of this?  The smtotc_16 implementation looked right, but is hard to follow all the bit math.

```suggestion
	
   // convert sign-magnitude value to 2's complement.   If bit 0x2000 is set this is a negative number, 
   //convert all the bits to the right of the sign bit to 2s complement
   if (rawCJReg & 0x2000)
   {
      rawCJReg = -(rawCJReg & 0x1FFF);
   }

```

Comment (Litemage) [reply src/PWFusion_MAX31856.cpp]: That's fair, those functions took me a while to work through (I lifted them from SO) 

I'll give this a test and report back

Comment (Litemage) [reply src/PWFusion_MAX31856.cpp]: So, it works, but the solution looks something more like this (this is for the LT registers):

```C++
   // convert sign-magnitude value to 2's complement.   If bit 0x40000 is set this is a negative number, 
   // convert all the bits to the right of the sign bit to 2s complement
   // Shift to align with memory
   rawTCReg >>= 13;
   if (rawTCReg & 0x40000)
   {
      // Negative
      rawTCTemp = (int32_t)(-(rawTCReg & 0x3FFFF));
   }
   else
   {
      // Positive
      rawTCTemp = (int32_t)(rawTCReg & 0x3FFFF);
   }
```

I agree that it is more readable, if you want to go this route. I have verified that this solution works, as well as it's 16-bit counterpart
